src/crypto-api/index.ts

```diff
@@ -995,15 +1000,77 @@ export interface CrossSigningStatus {
 /**
  * Crypto callbacks provided by the application
  */
-export interface CryptoCallbacks extends SecretStorageCallbacks {
+export interface CryptoCallbacks {
+    /**
+     * Called to retrieve a secret storage encryption key.
+     *
+     * [Server-side secret storage](https://spec.matrix.org/v1.12/client-server-api/#key-storage)
+     * is, as the name implies, a mechanism for storing secrets which should be shared between
+     * clients on the server. For example, it is typically used for storing the
+     * [key backup decryption key](https://spec.matrix.org/v1.12/client-server-api/#decryption-key)
+     * and the private [cross-signing keys](https://spec.matrix.org/v1.12/client-server-api/#cross-signing).
+     *
+     * The secret storage mechanism encrypts the secrets before uploading them to the server using a
+     * secret storage key. The schema supports multiple keys, but in practice only one tends to be used
+     * at once; this is the "default secret storage key" and may be known as the "recovery key" (or, sometimes,
+     * the "security key").
+     *
+     * Secret storage can be set up by calling {@link CryptoApi.bootstrapSecretStorage}. Having done so, when
+     * the crypto stack needs to access secret storage (for example, when setting up a new device, or to
+     * store newly-generated secrets), it will use this callback.
+     *
+     * Note that the secret storage key may be needed several times in quick succession: it is recommended
+     * that applications use a temporary cache to avoid prompting the user multiple times for the key. See
+     * also {@link cacheSecretStorageKey} which is called when a new key is created.
+     *
+     * The helper method {@link deriveRecoveryKeyFromPassphrase} may be useful if the secret storage key
+     * was derived from a passphrase.
+     *
+     * @param opts - An options object.
+     *
+     * @param name - the name of the *secret* (NB: not the encryption key) being stored or retrieved.
+     *    This is the "event type" stored in account data.
```

Comment:
I don't understand this, but that may be because I don't have enough expected context.

Reply:
"Secret storage" is really just "encrypted account data" (the spec, unhelpfully, refers to "account data" as "client config": https://spec.matrix.org/v1.12/client-server-api/#client-config). So, when I store a secret in secret storage, I encrypt it with the recovery key, and stash it as an account data "event". The `name` of the secret storage secret is then the `type` of the account data item.

Is this any better?

```suggestion
     *    When the item is stored in account data, it will have this `type`.
```

Reply:
Yes, that is better.

~~Maybe this is not the right place to explain it, but it's not clear to me how I would choose what argument I would need to pass in here.~~

But I've just realised that we are probably writing this documentation for someone who is *implementing* this method, not *calling* it, so this does provide the info they need. So this is good.